main.tf

```diff
@@ -83,6 +83,7 @@ resource "google_compute_instance_from_template" "bastion_vm" {
   network_interface {
     subnetwork         = var.subnet
     subnetwork_project = var.host_project != "" ? var.host_project : var.project
+    access_config = var.ephemeral_ip ? [{nat_ip="",network_tier="PREMIUM",public_ptr_domain_name=""}] : []
```

Comment:
Two comments:
1. I think you meant to pass in `var.network_tier` here instead of `"PREMIUM"`
2. If we're going to pass an access config to allow an ephemeral IP, let's allow users to edit all the values as a list using an `access_configs` variable. ie.

```
variable "access_configs" {
  type = list(object(
    network_tier = string
    nat_ip = string
    public_ptr_domain_name
  ))
  default = { 
    nat_ip=""
    network_tier="PREMIUM"
    public_ptr_domain_name=""
  }
}
```

Reply:
Ok yeah, that's a great idea. I will have a requirement for domain name soon so this solves that problem for me as well. I'll look into getting it done today.